ledger/src/blockstore.rs

```diff
@@ -1245,14 +1330,27 @@ impl Blockstore {
             }
         }
 
-        let erasure_meta_entry = erasure_metas.entry(erasure_set).or_insert_with(|| {
-            self.erasure_meta(erasure_set)
-                .expect("Expect database get to succeed")
-                .map(WorkingEntry::Clean)
-                .unwrap_or_else(|| {
-                    WorkingEntry::Dirty(ErasureMeta::from_coding_shred(&shred).unwrap())
-                })
-        });
+        let is_new_erasure_meta = match erasure_metas.entry(erasure_set) {
+            BTreeMapEntry::Vacant(entry) => {
+                let (erasure_meta, is_new_erasure_meta) = self
+                    .erasure_meta(erasure_set)
+                    .expect("Expect database get to succeed")
+                    .map(WorkingEntry::Clean)
+                    .map(|entry| (entry, false))
+                    .unwrap_or_else(|| {
+                        (
+                            WorkingEntry::Dirty(ErasureMeta::from_coding_shred(&shred).unwrap()),
+                            true,
+                        )
+                    });
+                entry.insert(erasure_meta);
```

Comment:
instead of `false/true` above, can't we just check if the output of `entry.insert(erasure_meta)` is clean or dirty?
it returns the inserted value:
https://doc.rust-lang.org/std/collections/btree_map/struct.VacantEntry.html#method.insert

Reply:
nice, that's much cleaner